keps/sig-network/2732-np-version/README.md

```diff
@@ -0,0 +1,803 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Because NetworkPolicy is implemented by components outside of the
+Kubernetes core, the existing versioning mechanisms in Kubernetes do
+not work well for it; instead of having a two-way version skew between
+"what features the user wants to use" and "what features the cluster
+supports", there is a three-way split between "what features the user
+wants to use", "what features the apiserver knows about", and "what
+features the NetworkPolicy implementation supports". Especially, there
+is often a lag between when new NetworkPolicy features are defined and
+when they are widely implemented. This means that in some cases a user
+cannot know if a cluster will correctly process a given NetworkPolicy
+(and makes it difficult to add new NetworkPolicy features if we want
+to ensure they won't be catastrophically misinterpreted by older
+implementations).
+
+```
+<<[UNRESOLVED snowflake ]>>
+
+Just how special/unusual is NetworkPolicy in this regard? Have any
+other SIGs already dealt with similar problems? Or if not are there
+any other APIs that are similarly failing to deal with the same
+problems?
+
+<<[/UNRESOLVED]>>
+```
+
+Somewhat related to this, there is currently no way for a
+NetworkPolicy implementation (ie, the network plugin) to indicate the
+status of NetworkPolicy processing to the user who created the policy.
+
+This KEP attempts to solve both of these problems.
+
+(Although it is not technically required that NetworkPolicy be
+implemented by the network plugin, this is almost universally the case,
+and so I will refer to NetworkPolicy implementations as "network
+plugins" through the rest of this document.)
+
+## Motivation
+
+### Goals
+
+- Provide a better mechanism for dealing with version skew with
+  respect to NetworkPolicy features. Specifically:
+
+    - Allow the apiserver to indicate to the network plugin what
+      version of the NetworkPolicy featureset a particular policy
+      requires (eg, whether it uses combined `podSelector` and
+      `namespaceSelector`), so that network plugins can reliably
+      recognize when a NetworkPolicy makes use of a feature that the
+      NetworkPolicy implementation does not support.
+
+    - Allow NetworkPolicy implementations to indicate back to the user
+      when they are aware that they cannot correctly implement a given
+      NetworkPolicy.
+
+- Define rules for dealing with feature gates and alpha APIs in
+  NetworkPolicy that work well with the 3-way versioning split.
+
+- Allow network plugins to indicate when a NetworkPolicy has been fully
```

Comment:
why not put those 2 items related to status into different KEP? It looks like not related to versioning.